contracts/integration/GelatoOnebalance/GelatoOnebalanceTarget.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {GelatoOneBalanceAdapter} from "./GelatoOnebalanceAdapter.sol";
+import {SwapForwarderXReceiver} from "../../destination/xreceivers/Swap/SwapForwarderXReceiver.sol";
+
+contract GelatoOneBalanceTarget is GelatoOneBalanceAdapter, SwapForwarderXReceiver {
+    constructor (address _connext, address _gelato1balance) SwapForwarderXReceiver(_connext) GelatoOneBalanceAdapter(_gelato1balance) {}
+
+    function _forwardFunctionCall(
+    bytes memory _preparedData,
+    bytes32 /*_transferId*/,
+    uint256 /*_amount*/,
+    address /*_asset*/
+  ) internal override returns (bool) {
+    (bytes memory _forwardCallData, uint256 _amountOut, , ) = abi.decode(
+      _preparedData,
+      (bytes, uint256, address, address)
+    );
+    (address _sponsor, address _token, uint256 _amount) = abi.decode(
+        _forwardCallData, 
+        (address, address, uint256)
+    );
+
+    depositTokens(_sponsor, _token, _amount);
```

Comment:
should we add event here? indexing sponsor, transferId, amount,

Reply:
so internally `depositTokens` method revert if it fails, so it will only return `true` when it passes, so Dont think we need an event here.

Reply:
event act like cheap storage, so it will help us track data offline going via these target contracts.

Do try/catch alternative with call() for `despositTokens()`, And then return success and log it in event as well.

Reply:
```
event LogDeposit(
        bytes32 indexed root,
        address indexed sponsor,
        address indexed token,
        uint256 amount
    );
```
This is already emitting underlying depositToken function, do we still need one?

Reply:
checked, we are handling it under the forwarder contract. But i still think it's better to do try/catch under forwardCallFunction and log there with indexing of encoding data as better data points.
cc: @rhlsthrm

Reply:
just for reference the logs we will be getting from xReceive functions contains `transferId` and `errorMessage`.

Reply:
I don't think so, we should be reducing the complexity of the integration and not require try-catch. The best practice is for lower level modules and functions to throw errors while higher-level modules catch it.